include/sound/sof/channel_map.h

```diff
@@ -0,0 +1,65 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Marcin Maka <marcin.maka@linux.intel.com>
+ *         Slawomir Blauciak <slawomir.blauciak@linux.intel.com>
+ */
+
+#ifndef __IPC_CHANNEL_MAP_H__
+#define __IPC_CHANNEL_MAP_H__
+
+#include <ipc/header.h>
+#include <stdint.h>
+
+/**
+ * \brief Channel map, specifies transformation of one-to-many or many-to-one.
+ *
+ * In case of one-to-many specifies how the output channels are computed out of
+ * a single source channel,
+ * in case of many-to-one specifies how a single target channel is computed
+ * from a multichannel input stream.
+ *
+ * Channel index specifies position of the channel in the stream on the 'one'
+ * side.
+ *
+ * Ext ID is the identifier of external part of the transformation. Depending
+ * on the context, it may be pipeline ID, dai ID, ...
+ *
+ * Channel mask describes which channels are taken into account on the "many"
+ * side. Bit[i] set to 1 means that i-th channel is used for computation
+ * (either as source or as a target).
+ *
+ * Channel mask is followed by array of coefficients in Q2.30 format,
```

Comment:
why Q2.30? there should be an explanation here on the fixed point math. @singalsu can you chime in?

Reply:
I recall this was discussed with @mmaka1. With Q2.30 we can cover mix gains in range 0 - 1.9999... If the fractional format would be Q1.31 we could not get exactly unity gain but something like 0.9999... We don't need amplification but due to need for unity gain the Q2.30 format was selected.

Reply:
yes, I suspected we didn't want amplification, so this needs to be made explicit.
I am not clear however if it's desirable to use unity gain when you mix multiple channels,  you have a risk of saturation and the loudness level will increase.